gammapy/spectrum/flux_point.py

```diff
@@ -18,12 +19,379 @@
+        ax = fig.add_subplot(111)
+        self.plot(ax=ax, **kwargs)
+        return ax
+
+    def __str__(self):
+        """
+        String representation of the flux points class.
+        """
+        info = ''
+        info += "Flux points of type '{}'".format(self.sed_type)
+        return info
+
+    def info(self):
+        """
+        Print flux points info.
+        """
+        print(self)
+
+    @classmethod
+    def read(cls, filename, **kwargs):
+        """
+        Read flux points.
+
+        Parameters
+        ----------
+        filename : str
+            Filename
+        kwargs : dict
+            Keyword arguments passed to `~astropy.table.Table.read`.
+
+        Examples
+        --------
+
+        >>> from gammapy.spectrum import FluxPoints
+        >>> filename = '$GAMMAPY_EXTRA/test_datasets/spectrum/flux_points/flux_points.fits'
+        >>> flux_points = FluxPoints.read(filename)
+        >>> flux_points.show()
+
+        """
+        filename = make_path(filename)
+        try:
+            table = Table.read(str(filename), **kwargs)
+        except IORegistryError:
+            kwargs.setdefault('format', 'ascii.ecsv')
+            table = Table.read(str(filename), **kwargs)
+
+        if 'SED_TYPE' not in table.meta.keys():
+            sed_type = cls._guess_sed_type(table)
+            table.meta['SED_TYPE'] = sed_type
+
+        return cls(table=table)
+
+    def write(self, filename, **kwargs):
+        """
+        Write flux points.
+
+        Parameters
+        ----------
+        filename : str
+            Filename
+        kwargs : dict
+            Keyword arguments passed to `~astropy.table.Table.write`.
+        """
+        try:
+            self.table.write(str(filename), **kwargs)
+        except IORegistryError:
+            kwargs.setdefault('format', 'ascii.ecsv')
+            self.table.write(str(filename), **kwargs)
+
+    # TODO: handle with Energy or EnergyBounds classes?
+    @property
+    def e_ref(self):
+        """
+        Reference energy.
+
+        Defined by `e_ref` column in `FluxPoints.table` or computed as log
+        center, if `e_min` and `e_max` columns are present in `FluxPoints.table`.
+
+        Returns
+        -------
+        e_ref : `~astropy.units.Quantity`
+            Reference energy.
+        """
+        try:
+            return self.table['e_ref'].quantity
+        except KeyError:
+            e_ref = np.sqrt(self.e_min * self.e_max)
+            return e_ref
+
+    # TODO: handle with Energy or EnergyBounds classes?
+    @property
+    def e_min(self):
+        """
+        Lower bound of energy bin.
+
+        Defined by `e_min` column in `FluxPoints.table`.
+
+        Returns
+        -------
+        e_min : `~astropy.units.Quantity`
+            Lower bound of energy bin.
+        """
+        try:
+            return self.table['e_min'].quantity
+        except KeyError:
+            raise NotImplementedError
```

Comment:
`KeyError` is better than `NotImplementedError` for something that isn't available.
Suggest to remove try-except here completely.
(and to not invent some `e_min` based on other columns)